src/http/data_loader.rs

```diff
@@ -16,10 +17,30 @@ use crate::json::JsonLike;
 pub struct HttpDataLoader {
   pub client: Arc<dyn HttpClient>,
   pub batched: Option<GroupBy>,
+  #[allow(clippy::type_complexity)]
+  pub get_body_value: Arc<dyn Fn(HashMap<String, Vec<&ConstValue>>, &Cow<str>) -> ConstValue + Send + Sync>,
 }
 impl HttpDataLoader {
-  pub fn new(client: Arc<dyn HttpClient>, batched: Option<GroupBy>) -> Self {
-    HttpDataLoader { client, batched }
+  pub fn new(client: Arc<dyn HttpClient>, batched: Option<GroupBy>, is_list: bool) -> Self {
+    let get_body_value: fn(HashMap<String, Vec<&ConstValue>>, &Cow<str>) -> ConstValue = match is_list {
+      true => |body_value: HashMap<String, Vec<&ConstValue>>, id: &Cow<str>| {
+        ConstValue::List(
+          body_value
+            .get(id.as_ref())
+            .unwrap_or(&Vec::new())
+            .iter()
+            .map(|&o| o.to_owned())
+            .collect::<Vec<_>>(),
+        )
+      },
+      false => |body_value: HashMap<String, Vec<&ConstValue>>, id: &Cow<str>| {
+        body_value
+          .get(id.as_ref())
+          .and_then(|a| a.first().cloned().cloned())
+          .unwrap_or(ConstValue::Null)
+      },
+    };
```

Comment:
have you considered to store the flag itself HttpDataLoader and call just usual functions based on that flag?

I expect that approach be more readable that doesn't involve closures and arc wrapper